src/exercism/ResistorColor.java

```diff
@@ -0,0 +1,26 @@
+class ResistorColor {    
+    String[] colorNames = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
+
+    int colorCode(String color) {
+        int code = -1;
+        for (int i = 0; i < colorNames.length; i++) {
+            if (colorNames[i].equals(color)) {
+                code = i;
+                return code;
```

Comment:
In your latest commit, you changed your solution. Previously you had "return code" AFTER the for loop.

Both solution are good.

Since you moved the "return code" inside your for loop, there is no need at all to have the local variable code. You can remove it and just have "return i"